src/fft.jl

```diff
@@ -277,6 +277,34 @@ for P in (:cFFTWPlan, :rFFTWPlan, :r2rFFTWPlan) # complex, r2c/c2r, and r2r
     end
 end
 
+mutable struct r2rFFTWPlan{T<:fftwNumber,K,inplace,N,G} <: FFTWPlan{T,K,inplace}
+    plan::PlanPtr
+    sz::NTuple{N,Int} # size of array on which plan operates (Int tuple)
+    osz::NTuple{N,Int} # size of output array (Int tuple)
+    istride::NTuple{N,Int} # strides of input
+    ostride::NTuple{N,Int} # strides of output
+    ialign::Int32 # alignment mod 16 of input
+    oalign::Int32 # alignment mod 16 of input
+    flags::UInt32 # planner flags
+    region::G # region (iterable) of dims that are transormed
+    kinds::K
+    pinv::ScaledPlan
```

Comment:
Should we parameterize this?  `ScaledPlan` is an abstract type, no?

Reply:
Yes, it is, but it's a bit tricky to do this properly, as the struct is initialized incompletely. Maybe I'll look into this in a separate PR